sdcm/utils/data_validator.py

```diff
@@ -309,7 +309,7 @@ def get_view_name_from_profile(self, name_substr, cs_profile=None, all_entries=F
                     if not all_entries:
                         break
 
-        return mv_names
+        return list(dict.fromkeys(mv_names))  # list of unique view names and keep the elements order
```

Comment:
why are you using `dict.fromkeys` and not `set` ? does it behave any differently ?

Reply:
`set` does not keep the element order (this was the original problem, that caused the failure). `dict.fromkeys` keeps the order